draft-thomson-http-replay.md

```diff
@@ -287,6 +289,37 @@ forwarding the request until the TLS handshake completes, or send a 4NN (Too
 Early) status code in response.  A gateway that is uncertain about whether an
 origin server supports the `Early-Data` header field SHOULD disable early data.
 
+## Consistent Handling of Early Data {#be-consistent}
+
+Consistent treatment of a request that arrives in - or partially in - early data
+is critical to avoiding inappropriate processing of replayed requests.  Requests
+that might be processed prior to the handshake completing MUST be processed the
+same way, even if the handshake is complete when processing occurs.
+
+A TLS implementation MUST ensure that it is possible to correctly identify early
+data.  This is necessary even if early data arrives at the same time as the TLS
+Finished message from the client.  If early data is misclassified, requests
+might erroneously appear to be safe to handle.  Even though the handshake might
+complete successfully on this connection, this is no guarantee that the same
+early data was not processed prior to handshake completion on another
+connection where the early data was replayed.
```

Comment:
Is this last sentence dependent upon the previous ones, or is it a separate point that deserves its own paragraph? It seems to me that this is an important bit of guidance to give to servers when they're deciding whether to accept early data, but it's buried in a paragraph that starts with requirements about TLS implementations.